### PR TITLE
Recursive dir

### PR DESCRIPTION
**Pull request by Balamurugan35**
Recursive dir

### CODE REVIEW

File: semantic_ai/connectors/microsoft/sharepoint.py [hunk 1]
```diff
@@ -14,6 +16,16 @@
 logger = logging.getLogger(__name__)
 
 
+async def get_path(url):
+    index = url.find('PowerAppsProd')
```
Comment (Tamil-Arasan-31) [inline semantic_ai/connectors/microsoft/sharepoint.py:20]: couldn't able to analyse what is happening here

**Review by Tamil-Arasan-31 — COMMENTED** [review]
good to go

File: semantic_ai/connectors/microsoft/sharepoint.py [hunk 2]
```diff
@@ -14,6 +16,16 @@
...
+    if index != -1:
+        index = url.find('PowerAppsProd')
```
Comment (SyedHabibM) [inline semantic_ai/connectors/microsoft/sharepoint.py:22]: What is this?!

File: semantic_ai/connectors/microsoft/sharepoint.py [hunk 3]
```diff
@@ -54,7 +66,7 @@ async def __make_request(self, url):
             'Authorization': 'Bearer {}'.format(await self.__get_access_token())
         }
         async with httpx.AsyncClient() as cli:
-            r = await cli.get(url, headers=headers)
+            r = await cli.get(url, headers=headers, timeout=60)
```
Comment (SyedHabibM) [inline semantic_ai/connectors/microsoft/sharepoint.py:69]: By default, this would be `30`. `60` is more.

**Review by SyedHabibM — CHANGES_REQUESTED** [review]

File: semantic_ai/connectors/microsoft/sharepoint.py (hunk 2 above)
Comment (Balamurugan35) [reply semantic_ai/connectors/microsoft/sharepoint.py:22]: to get the folderurl

File: semantic_ai/connectors/microsoft/sharepoint.py (hunk 3 above)
Comment (Balamurugan35) [reply semantic_ai/connectors/microsoft/sharepoint.py]: changed

**Review by SyedHabibM — APPROVED** [review]

**Pull request merged**